lib/ansible/modules/windows/win_domain.ps1

```diff
@@ -2,7 +2,7 @@
 
 # GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
 
-#Requires -Module Ansible.ModuleUtils.Legacy
+#AnsibleRequires -CSharpUtil Ansible.Basic
```

Comment:
This is not only a feature PR since you actually changed the module to use the new C# utils.
This should be a different PR, one for the utils and one for the feature.
I know it sounds like a lot but it will make the reviews faster and be the better way to add all of the changes into the module.